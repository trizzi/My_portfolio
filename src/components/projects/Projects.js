import React from 'react';
import projects from './projects.json';
import ProjectItem from './ProjectItem';
import projectImage from '../../assets/projects_background_image.svg';
import projectLogo from '../../assets/project_icon.svg';

const Projects = () => {
	return (
		<div
			name='section3'
			className='container-xl'
			style={{
				backgroundImage: `url(${projectImage})`,
				backgroundRepeat: 'no-repeat',
				backgroundSize: 'cover',
				// height: '100vh',
				backgroundColor: 'rgba(0,0,0,0.9)',
			}}>
			<div>
				<div className='flex px-12 pt-8'>
					<div>
						<img
							src={projectLogo}
							alt='service-logo'
						/>
					</div>
					<div className='pt-3'>
						<h1>Projects</h1>
					</div>
				</div>
			</div>

			<div className='container mx-auto p-8 lg:p-20'>
				<div className='flex flex-wrap gap-4'>
					{projects.map((project) => (
						<ProjectItem
							key={project.id}
							project={project}
						/>
					))}
				</div>
			</div>
		</div>
	);
};

export default Projects;
