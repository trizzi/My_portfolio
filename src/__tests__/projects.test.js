import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Projects from '../components/projects/Projects';
import { ThemeContextProvider } from '../context/ThemeContext';

// Mock the IntersectionObserver
beforeEach(() => {
	window.IntersectionObserver = jest.fn(function () {
		this.observe = jest.fn();
		this.disconnect = jest.fn();
		this.unobserve = jest.fn();
	});
});

// Mock the projects data
jest.mock('../components/projects/projects.json', () => [
	{ id: 1, title: 'Project 1', description: 'Description 1' },
	{ id: 2, title: 'Project 2', description: 'Description 2' },
	{ id: 3, title: 'Project 3', description: 'Description 3' },
]);

test('renders Projects section with correct content', () => {
	const { container } = render(
		<ThemeContextProvider>
			<Projects />
		</ThemeContextProvider>,
	);

	// Check if the projects section is rendered
	const projectsSection = screen.getByTestId('project-section');
	expect(projectsSection).toBeInTheDocument();

	// Check if the project items are rendered with the correct data
	const projectTitles = screen.getAllByRole('heading', { level: 3 }); // Assuming project titles are rendered as h3 elements
	expect(projectTitles).toHaveLength(3); // Assuming there are three projects

	// const projectDescriptions = screen.getAllByText(/Description/i);
	// // Matches any text containing "Description"
	// expect(projectDescriptions).toHaveLength(3); // Assuming there are three projects

	expect(container).toMatchSnapshot();
});
