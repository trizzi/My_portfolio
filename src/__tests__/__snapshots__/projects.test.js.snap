// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Projects section with correct content 1`] = `
<div>
  <div
    class="px-4 py-10"
    data-testid="project-section"
    name="section3"
    style="background-image: url(projects_background_image.svg); background-repeat: no-repeat; background-size: cover; background-color: rgba(0, 0, 0, 0.9);"
  >
    <div
      class="container-xl lg:container m-auto"
    >
      <div>
        <div
          class="flex justify-center "
        >
          <div>
            <img
              alt="service-logo"
              class="mx-auto h-24 rounded-full sm:mx-0 sm:shrink-0"
              src="project_icon.svg"
            />
          </div>
          <h1
            class="text-4xl text-bold pb-6"
          >
            Projects
          </h1>
        </div>
      </div>
      <div
        class=""
      >
        <div
          class="grid grid-cols-1 md:grid-cols-3 gap-8"
        >
          <div
            class="w-full mt-10 lg:mt-0 lg:mr-0 md:w-2/5 lg:w-[400px] rounded-xl overflow-hidden shadow-xl"
            style="opacity: 0; transform: translateY(100px) translateZ(0);"
          >
            <img
              alt="project_image"
              class="rounded-t-lg w-full"
            />
            <div
              class="pt-5 pb-10 px-10 bg-[#292A3C] opacity-80 rounded-xl"
            >
              <h3
                class="font-bold text-xl py-5"
              />
              <p />
              <span
                class="whitespace-nowrap flex items-center pt-2"
              >
                <a>
                  <button
                    class="
			rounded-lg py-2 lg:py-2 px-5 whitespace-nowrap hover:bg-violet-400
        undefined undefined mt-5"
                    style="background-color: rgb(153, 50, 204); color: rgb(255, 255, 255); border: 2px solid #9932cc; opacity: 0;"
                    tabindex="0"
                  >
                    Explore
                  </button>
                </a>
              </span>
            </div>
          </div>
          <div
            class="w-full mt-10 lg:mt-0 lg:mr-0 md:w-2/5 lg:w-[400px] rounded-xl overflow-hidden shadow-xl"
            style="opacity: 0; transform: translateY(100px) translateZ(0);"
          >
            <img
              alt="project_image"
              class="rounded-t-lg w-full"
            />
            <div
              class="pt-5 pb-10 px-10 bg-[#292A3C] opacity-80 rounded-xl"
            >
              <h3
                class="font-bold text-xl py-5"
              />
              <p />
              <span
                class="whitespace-nowrap flex items-center pt-2"
              >
                <a>
                  <button
                    class="
			rounded-lg py-2 lg:py-2 px-5 whitespace-nowrap hover:bg-violet-400
        undefined undefined mt-5"
                    style="background-color: rgb(153, 50, 204); color: rgb(255, 255, 255); border: 2px solid #9932cc; opacity: 0;"
                    tabindex="0"
                  >
                    Explore
                  </button>
                </a>
              </span>
            </div>
          </div>
          <div
            class="w-full mt-10 lg:mt-0 lg:mr-0 md:w-2/5 lg:w-[400px] rounded-xl overflow-hidden shadow-xl"
            style="opacity: 0; transform: translateY(100px) translateZ(0);"
          >
            <img
              alt="project_image"
              class="rounded-t-lg w-full"
            />
            <div
              class="pt-5 pb-10 px-10 bg-[#292A3C] opacity-80 rounded-xl"
            >
              <h3
                class="font-bold text-xl py-5"
              />
              <p />
              <span
                class="whitespace-nowrap flex items-center pt-2"
              >
                <a>
                  <button
                    class="
			rounded-lg py-2 lg:py-2 px-5 whitespace-nowrap hover:bg-violet-400
        undefined undefined mt-5"
                    style="background-color: rgb(153, 50, 204); color: rgb(255, 255, 255); border: 2px solid #9932cc; opacity: 0;"
                    tabindex="0"
                  >
                    Explore
                  </button>
                </a>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
